### PR TITLE
Replace generated docs with a client-side React app

### PR DESCRIPTION
**Pull request by bhousel**
This PR replaces the generated static html doc pages with a React.js app that runs in the client.  It looks exactly the same as the existing doc pages.

The app pulls data from the files in `dist/` so we won't need to generate a bunch of doc pages anymore. 

It also paves the way for some useful features like client-side filtering mentioned in #2734, #2733, #2438

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ in OpenStreetMap.
 ### Browse the index
 
 You can browse the index at
-http://osmlab.github.io/name-suggestion-index/brands/index.html
+http://osmlab.github.io/name-suggestion-index/brands
```
Comment (tas50) [inline README.md]: this is a 404

Comment (bhousel) [reply README.md]: Yeah I see that now - gh-pages isn't quite compatible with react router, so I need to change a bit how the app works.  I'll fix it tonight..

Comment (bhousel) [reply README.md]: It works now!  I switched from using path-based routing to using querystrings instead..

Comment (Adamant36) [reply README.md]: It appears that clicking on any of the links just loops back to the main brand page.

Comment (bhousel) [reply README.md]: Thanks for reporting @Adamant36  -  I fixed it in 2ec4117f0, but let me know if you find any other issues.

**Pull request merged**